src/irs/core/storage_inspector.py

```diff
@@ -65,25 +65,35 @@ async def check_buckets(self):
             for object_id in await object_storage.list_all_object_ids(bucket_id):
                 try:
                     staging_object = await self._staging_object_dao.find_one(
-                        mapping={"object_id": object_id}
+                        mapping={"object_id": object_id, "storage_alias": storage_alias}
                     )
                 except NoHitsFoundError:
+                    # This can happen in one of two cases:
+                    # 1) Failed to send success event
+                    # 2) Success event sent, but failed to insert staging object entry
                     extra = {
                         "object_id": object_id,
                         "bucket_id": bucket_id,
                         "storage_alias": storage_alias,
                     }
-
                     log.error(
                         "Object '%s' with no corresponding DB entry found in bucket '%s'"
                         + " of storage '%s'.",
                         *extra.values(),
                         extra=extra,
                     )
                     continue
-
-                if staging_object.storage_alias != storage_alias:
-                    # can't look up bases on two fields, skip for now and log later
+                except MultipleHitsFoundError as error:
```

Comment:
This should also be raised, right? We can also leave it out again as you suggested. Sorry for the confusion.

Reply:
My thought here was to just go ahead with logging and see if we have more than just one problem and this might point to some systemic issues. Either way, it's just the separate CLI command and it's not changing any state, so no harm in letting it finish.

Reply:
Right, let's do so.